cpp/examples/arrow/CMakeLists.txt

```diff
@@ -42,8 +42,12 @@ if(ARROW_FLIGHT)
     message(FATAL_ERROR "Statically built gRPC requires ARROW_BUILD_STATIC=ON")
   else()
     set(FLIGHT_EXAMPLES_LINK_LIBS arrow_flight_static)
-    set(GRPC_REFLECTION_LINK_LIBS -Wl,--whole-archive gRPC::grpc++_reflection
-                                  -Wl,--no-whole-archive)
+    if(APPLE)
+      set(GRPC_REFLECTION_LINK_LIBS -Wl,-force_load gRPC::grpc++_reflection -Wl)
```

Comment:
Shouldn't this end with `-Wl,--no-force-load` or something?

Related, does it take `-force-load` or `--force-load`?

Reply:
So `all_load`/`noall_load` currently won't work due to symbol duplication so `force-load` is the only option and there appears to be no `noforce-load` option.
More [here](https://github.com/premake/premake-core/issues/1170).